homeassistant/components/generic/camera.py

```diff
@@ -65,24 +66,52 @@
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up a generic IP Camera."""
 
-    await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
+    _LOGGER.warning(
+        "Loading generic IP camera via configuration.yaml is deprecated, "
+        "it will be automatically imported.  Once you have confirmed correct "
+        "operation, please remove 'generic' (IP camera) section(s) from "
+        "configuration.yaml"
+    )
+    streamsource = config.get(CONF_STREAM_SOURCE)
+    config[CONF_STILL_IMAGE_URL] = config[CONF_STILL_IMAGE_URL].template
+    config[CONF_STREAM_SOURCE] = (
+        streamsource.template if streamsource is not None else None
+    )
```

Comment:
```suggestion
    stream_source = config.get(CONF_STREAM_SOURCE)
    config[CONF_STILL_IMAGE_URL] = config[CONF_STILL_IMAGE_URL].template
    config[CONF_STREAM_SOURCE] = (
        stream_source.template if stream_source is not None else None
    )
```
To be consistent with the config flow

Reply:
Agreed, done.